### PR TITLE
Reporting landcover

### PR DESCRIPTION
**Pull request by myarjunar**
### What does it fix?
* Ticket: #3626 
* Funded by: DFAT
* Description: Add report for estimated land-cover productivity loss based on #3626. This report can be generic for any other type of exposure. 
selection_112

cc @timlinux @Charlotte-Morgan @fredychandra

### CODE REVIEW

File: safe/report/extractors/analysis_detail.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # coding=utf-8
 """Module used to generate context for analysis detail section."""
-from safe.definitions.exposure import exposure_all, exposure_population
+from safe.definitions.exposure import exposure_all, exposure_population, \
```
Comment (ismailsunni) [inline safe/report/extractors/analysis_detail.py]: __[flake8]__
* [F401] 'safe.definitions.exposure.exposure_land_cover' imported but unused

**Comment by samnawi** [comment]
hi @myarjunar I've got a problem with your PR
```
Problem
The following problem(s) were encountered whilst running the analysis.

TypeError
Suggestion
You can try the following to resolve the issue:

Check that you have the latest version of InaSAFE installed - you may have encountered a bug that is fixed in a subsequent release.
Check the InaSAFE documentation to see if you are trying to do something unsupported.
Report the problem using the issue tracker at https://github.com/inasafe/inasafe/issues. Reporting an issue requires that you first create a free account at http://github.com. When you report the issue, please copy and paste the complete contents of this panel into the issue to ensure the best possible chance of getting your issue resolved.
Try contacting one of the InaSAFE development team by sending an email to info@inasafe.org. Please ensure that you copy and paste the complete contents of this panel into the email.
Details
These additional details were reported when the problem occurred.

unsupported operand type(s) for +=: 'int' and 'QPyNullVariant'
Diagnostics

In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1157, in run self._run()
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 122, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1351, in _run self.summary_calculation()
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 122, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_function\impact_function.py", line 1980, in summary_calculation self._aggregate_hazard_impacted, self._exposure_summary)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 122, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\gis\vector\summary_4_exposure_summary_table.py", line 181, in exposure_summary_table summarization_dicts = summarize_result(exposure_summary, callback)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\utilities\profiling.py", line 122, in with_profiling ret = fn(*args, **kwargs)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\gis\vector\summary_4_exposure_summary_table.py", line 310, in summarize_result summarizer_field['field_name']]
```

**Comment by Gustry** [comment]
@samnawi Can you replicate this issue in the develop branch? or only with this PR?
Because, AFAIK, @myarjunar didn't touch the impact function too much in this PR, only the reporting side. And you error comes from before the reporting component.

**Comment by samnawi** [comment]
mmm.... it's strange @Gustry last week I tried the land cover analysis, and it can run well. But, today I get the same problem like above when I run landcover analysis in develop mode. I'll check with the other.

**Comment by samnawi** [comment]
image

:+1:  It's working now....

**Comment by myarjunar** [comment]
can I merge this PR? @samnawi @fredychandra

File: safe/definitions/fields.py [hunk 2]
```diff
@@ -2234,6 +2238,7 @@
     'key': 'affected_productivity_cost_field',
     'name': tr('Affected Productivity Cost'),
     'field_name': 'affected_productivity_cost',
+    'header_name': tr('Production Cost'),
```
Comment (Gustry) [inline safe/definitions/fields.py]: Production or productivity ?

File: safe/definitions/fields.py [hunk 3]
```diff
@@ -2307,6 +2313,7 @@
     'key': 'affected_productivity_value_field',
     'name': tr('Affected Productivity Value'),
     'field_name': 'affected_productivity_value',
+    'header_name': tr('Production Value'),
```
Comment (Gustry) [inline safe/definitions/fields.py]: Same question here

File: safe/definitions/fields.py (hunk 2 above)
Comment (myarjunar) [inline safe/definitions/fields.py]: based on @fredychandra example on #3626 , it's 'Production'. CMIIW
selection_117

Comment (Gustry) [inline safe/definitions/fields.py]: So why it's written `affected_productivity_cost` ? CF @ismailsunni with the discussion from this morning on the inasafe channel

Comment (ismailsunni) [inline safe/definitions/fields.py]: I think it's my fault. I will make a PR to make sure they are uniform. And the diagram that I said this morning.

**Comment by ismailsunni** [comment]
Hi @gustry and @myarjunar 
Let's merge this one first since the PR is more about the report.. I will update the definition of productivity based on the discussion in gitter.

**Comment by Gustry** [comment]
ok. I don't want to mix `productivity` and `production` in the same field definition.

Thanks @ismailsunni and @myarjunar

**Comment by ismailsunni** [comment]
@Gustry yes, me too. I just realized it after reading your comment.

**Pull request merged**